GMG/GMG/UI/Components/Metronome.swift

```diff
@@ -6,13 +6,149 @@
+        do {
+            try AVAudioSession.sharedInstance().setCategory(.playback, mode: .default, options: [.mixWithOthers])
+            try AVAudioSession.sharedInstance().setActive(true)
+        } catch {
+            print("[Metronome] AudioSession error: \(error)")
+        }
+    }
+
+    //소리 플레이어 (소리 구분)
+    private func loadPlayers() {
+        
+        if let url = Bundle.main.url(forResource: "Metronome_High", withExtension: "wav") {
+            highPlayer = try? AVAudioPlayer(contentsOf: url)
+            highPlayer?.volume = 1.0
+            highPlayer?.prepareToPlay()
+        } else {
+            print("[Metronome] Missing resource: Metronome_High.wav (check target membership)")
+        }
+        
+        if let url = Bundle.main.url(forResource: "Metronome_Low", withExtension: "wav") {
+            lowPlayer = try? AVAudioPlayer(contentsOf: url)
+            lowPlayer?.volume = 1.0
+            lowPlayer?.prepareToPlay()
+        } else {
+            print("[Metronome] Missing resource: Metronome_Low.wav (check target membership)")
+        }
+    }
+
+    // 소리 중 강박을 구분을 위한 인텍스 설정 (3/4, 4/4 박자는 첫 번째 박만 강박이지만, 6/8 박자의 경우 1, 4번 박에 강박을 줘야 함) *강박: Metronome 재생 중 기준을 잡는 높은 음
+    private func shouldUseHigh(beatIndex: Int) -> Bool {
+        if selectedTimeSig.numerator == 6 && selectedTimeSig.denumerator == .eighth {
+            return beatIndex == 0 || beatIndex == 3
+        } else {
+            return beatIndex == 0
+        }
+    }
+
+    // 위에서 구분한 Index에 따라 박자별 High or Low 음성 재생
+    private func playClick(for beatIndex: Int) {
+        if shouldUseHigh(beatIndex: beatIndex) {
+            guard let audioPlayer = highPlayer else {
+                print("[Metronome] highPlayer is nil")
+                return
+            }
+            audioPlayer.currentTime = 0
+            audioPlayer.play()
+        } else {
+            guard let audioPlayer = lowPlayer else {
+                print("[Metronome] lowPlayer is nil")
+                return
+            }
+            audioPlayer.currentTime = 0
+            audioPlayer.play()
+        }
+    }
+
     var body: some View {
-        Text(/*@START_MENU_TOKEN@*/"Hello, World!"/*@END_MENU_TOKEN@*/)
+        
+        //TimelineView를 사용해 beatInterval마다 뷰가 업데이트 되도록 구성함
+        TimelineView(.periodic(from: .now, by: beatInterval)) { context in
+            BeatBar(beatsPerBar: beatsPerBar, currentBeat: currentBeatToShow)
+            .animation(.spring(response: 0.28, dampingFraction: 0.85, blendDuration: 0.2), value: beatsPerBar)
+            .padding()
+            .onChange(of: context.date) { _, _ in
+                guard isPlaying, let start = startDate else { return }
+                let step = Int(floor(Date().timeIntervalSince(start) / beatInterval))
+                if step != lastStepPlayed {
+                    lastStepPlayed = step
+                    playClick(for: currentBeatToShow)
+                }
+            }
```

Comment:
modifier 들여쓰기 넣어주면 더 읽기 편할 것 같아요!
```suggestion
            BeatBar(beatsPerBar: beatsPerBar, currentBeat: currentBeatToShow)
                .animation(.spring(response: 0.28, dampingFraction: 0.85, blendDuration: 0.2), value: beatsPerBar)
                .padding()
                .onChange(of: context.date) { _, _ in
                    guard isPlaying, let start = startDate else { return }
                    let step = Int(floor(Date().timeIntervalSince(start) / beatInterval))
                    if step != lastStepPlayed {
                        lastStepPlayed = step
                        playClick(for: currentBeatToShow)
                    }
                }
```